src/third_party/blake3/CMakeLists.txt

```diff
@@ -115,13 +115,15 @@ _add_blake3_source_if_enabled(sse41 "" "-msse4.1" "_mm_test_all_ones(_mm_set1_ep
 _add_blake3_source_if_enabled(avx2 "/arch:AVX2" "-mavx2" "_mm256_abs_epi8(_mm256_set1_epi32(42))")
 _add_blake3_source_if_enabled(avx512 "/arch:AVX512" "-mavx512f -mavx512vl" "_mm256_abs_epi64(_mm256_set1_epi32(42))")
 
-# Neon is always available on AArch64
+# Neon is always available on AArch64, but blake3_neon.c only supports little-endian
 if(CMAKE_SIZEOF_VOID_P EQUAL 8)
   # https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics
   check_c_source_compiles(
     [=[
       #include <arm_neon.h>
+      #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
       int main() { vdupq_n_s32(42); return 0; }
+      #endif
```

Comment:
Maybe I'm missing something, but doesn't this always compile correctly if `__BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__` so that `HAVE_NEON` will become true?

Reply:
I have tested this approach on aarch64_be, it fails the HAVE_NEON test as intended